### DISCUSSION TITLE
os.cp如何将子文件夹里的内容按目录格式复制到根目录中

### DISCUSSION THREAD

**Discussion by Elite-stay** [xmake-io/xmake#5789] [Q&A]
我做了个模拟工程，里面有一次提交，按预期结果是`configs/config1.ini`和`doc/doc.md`被替换，其他文件不变，不知道咋整

[cptest.zip](https://github.com/user-attachments/files/17625471/cptest.zip)

**Comment by waruqi** [comment]
> 按目录格式复制到根目录中 

现在的配置已经按目录格式复制了，没啥问题。。

> 其他文件不变，不知道咋整

目前 os.cp 不支持增量复制，会整个目录替换复制。目前你只能自己遍历复制

**Reply by Elite-stay** [reply]
估计后面得提供一个，自己写的普通文件可以，符号链接不知道是不是好像有点问题
